src/job/jobHandler.ts

```diff
@@ -504,12 +544,65 @@ export abstract class JobHandler {
     return await this.executeBuild();
   }
 
+  /* New build process without Makefiles */
+  @throwIfJobInterupted()
+  protected async build(): Promise<boolean> {
+    this.cleanup();
+    const job = this._currJob;
+    const logger = (msg: string) => this.logger.save(this.currJob._id, msg);
```

Comment:
I talked to @casthewiz about the real-time logging, and he brought up a valid concern that we could potentially run into issues with the number of writes to the database. `this.logger.save` writes to the database every time, and so we might want to avoid this. 

Here's what I think we can do: First, call `logger` on the final result of the job i.e. we just log the resulting CLI output at once. It would look like this:

```ts

const logger = (output: CliCommandResponse) => {
  this._logger.save(this.currJob._id, parseResults.outputText);
  this._logger.save(this.currJob._id, parseResults.errorText);

}
const parseResults = await nextGenParse({ job });

```

We'll also want to maintain the real-time logging which you have added for sure! I'll address that specifically in a comment for the `executeCliCommand` function. It's a change that I should have added initially.